hybrasyl/Objects/Creature.cs

```diff
@@ -801,6 +805,14 @@ public virtual void RegenerateMp(double mp, Creature regenerator = null)
 
         public virtual void Damage(double damage, Enums.Element element = Enums.Element.None, Enums.DamageType damageType = Enums.DamageType.Direct, Castables.DamageFlags damageFlags = Castables.DamageFlags.None, Creature attacker = null, bool onDeath = true)
         {
+            if (attacker is User)
+            {
+                if (FirstHitter == null || !Game.World.ActiveUsersByName.ContainsKey(FirstHitter.Name) || ((DateTime.Now - LastHitTime).TotalSeconds > Constants.MONSTER_TAGGING_TIMEOUT)) FirstHitter = attacker;
+                if (attacker != FirstHitter && !((FirstHitter as User).Group?.Members.Contains(attacker) ?? false)) return;
```

Comment:
It occurs to me this logic will have unexpected results for certain edge cases (pvp, for one). For now, what do you think about updating it to `if (attacker is User && this is Monster)?`

Reply:
Makes sense to me. Will add now.